make themes: 2 node_modules directories instead of 3

dalf opened the pull request:
## Initial notes
* To merge after #150
* Currently look at the last two commits
* Only https://github.com/searxng/searxng/pull/150/commits/9108c90f0e9d17a8a14b502be3d33d129c93d294 is required from #150

## What does this PR do?

```make themes``` creates three node_modules directory:
* `.`
* `./searx/statics/themes/oscar`
* `./searx/statics/themes/simple`

The first one (at the root) contains dependencies described here:
https://github.com/searxng/searxng/blob/fdc367d58b3be8a54c3aaac3f2e873429c7ab040/manage#L16

This is not desirable for different reasons:
* `npm audit` doesn't work
* dependabot can't update the [dependencies](https://github.com/dependabot)
* `grunt` has to be called in a not standard way.

A second commit changes `node.clean` : it calls `npm run clean` for each themes.

## Why is this change important?

simpler build phases

## How to test this PR locally?

* `make themes`
* `make themes.ocar`
* `make node.clean`

## Author's checklist

<!-- additional notes for reviewiers -->

## Related issues

Related to:
* #150 
* #137
* https://github.com/searxng/searxng/wiki/Milestones#milestone-19---data-upgrade--build-process

dalf committed:
[mod] simple theme: update dependencies

dalf committed:
Static build

dalf committed:
[mod] simple theme: move source files to the src directory

dalf committed:
[mod] simple theme: leaflet becomes a packages.json dependency

the build of the themes updates:
* js/leaflet.js ( was leaflet/leaflet.js )
* css/leaflet.css ( was leaflet/leaflet.css )
* css/images ( was leaflet/images )

dalf committed:
[mod] simple theme: move magnet.svg to src/fonts

see https://github.com/searx/searx/pull/1109

dalf committed:
[mod] simple theme: autocomplete-js becomes a packages.json dependency

Update to version 2.7.1

dalf committed:
[mod] simple theme: remove src/less/ion.less

This file is generated by webfont.
* It is now generated as searx/static/themes/simple/ion.less
* It is generated before the .less compilation.
* .gitignore includes this file

Add two new package depedencies: fontforge ttfautohint
See utils/searx.sh

dalf committed:
Static build

dalf committed:
[mod] remove common grunt install

Before this commit, there are 3 node_modules directory:
* one in .
* two others in ./searx/statics/themes/*

This is no desirable:
* it declares the npm depdenencies in the shell script.
* dependabot can't updates theses dependencies.
* this is a not standard way to build a package (two different locations for the dependencies).

With this commit and the PR  #150 there is one unique node_modules directory per theme.

dalf committed:
[mod] make node.clean: call the "clean" script from the packages.json

The dependencies and build process are described in the gruntfiles.js and packages.json files.
It makes sense to move the clean up to these files too.

return42 commented on manage:
```diff
@@ -333,6 +333,7 @@ node.clean() {
        ./searx/static/themes/oscar/node_modules \
        ./searx/static/themes/simple/package-lock.json \
        ./searx/static/themes/simple/node_modules
+       ./searx/static/themes/simple/ion.less \
```
This is a (backslash) typo, but the `node.clean` command is fixed in 14ba56f

return42 committed:
[enh] ./manage node.env - check build tools first

The node.env build environment require npm, ttfautohint and fontforge installed
in the OS.  These tools can be installed by::

    sudo -H ./utils/searx.sh install buildhost

If one of the tools is not installed, the script node.env stops with a
appropriate message.

BTW: We ignore CentOS-7 as developer & build platform

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

dalf commented on utils/searx.sh:
```diff
@@ -88,7 +95,8 @@ BUILD_PACKAGES_centos="\
 texlive-collection-latex dejavu-sans-fonts dejavu-serif-fonts
-dejavu-sans-mono-fonts fontforge ttfautohint"
+dejavu-sans-mono-fonts fontforge ttfautohint
+"
```
No npm for this one ?

return42 replied:
These are the yum packages for cenos-7 (we have support only for centos-7). Modern versions of centos, fedora and RHEL are using dnf package manager. Automation of the yum package installation is more complicate and centos-7 was old from the beginning when we added it.  I vote to drop these old distribution.

Comiting the newline was a fail of mine / I will remove the newline. 

**update** In this context, please have a look at my commit ae677cb6 / is this commit OK for you?

return42 commented on utils/searx.sh:
```diff
@@ -88,7 +88,7 @@ BUILD_PACKAGES_centos="\
 texlive-xetex-bin texlive-collection-fontsrecommended
 texlive-collection-latex dejavu-sans-fonts dejavu-serif-fonts
-dejavu-sans-mono-fonts"
+dejavu-sans-mono-fonts fontforge ttfautohint"
```
I haven't tested centos-7 (yum package manager), I don't know if this works .. see https://github.com/searxng/searxng/pull/151#discussion_r653031106

Lets ignore CentOS-7 as developer (build) platform.

**update** In this context, please have a look at my commit ae677cb6 / is this commit OK for you?

return42 commented on searx/static/themes/simple/package.json:
```diff
@@ -20,8 +20,8 @@
-    "build": "npm install && grunt",
-    "start": "grunt watch",
-    "test": "grunt"
+    "all": "npm install && grunt",
```
For my understanding: is `all` the default _target_ when calling `npm install`?

dalf replied:
See https://docs.npmjs.com/cli/v7/commands/npm-run-script
There is no default command, but there are some shortcuts:
* `npm start` ==> `npm run start`
* same for `restart`, `stop`, `test`.

In package.json of this PR:
* `npm run all` : download dependencies, build the theme
* `npm run build`: build the theme
* `npm run watch`: same spirit than `make docs-live` but for this theme instead of the documentation : any change to the source files recompile the theme ( https://github.com/gruntjs/grunt-contrib-watch )

return42 reviewed (approved):
I finished my review and placed ae677cb6 on top of this branch.

FY: I won't review #150 since it is already a part of this PR #151.
